l1-contracts/test/NewOutbox.t.sol

```diff
@@ -0,0 +1,254 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2024 Aztec Labs.
+pragma solidity >=0.8.18;
+
+import {Test} from "forge-std/Test.sol";
+import {NewOutbox} from "../src/core/messagebridge/NewOutbox.sol";
+import {INewOutbox} from "../src/core/interfaces/messagebridge/INewOutbox.sol";
+import {Errors} from "../src/core/libraries/Errors.sol";
+import {DataStructures} from "../src/core/libraries/DataStructures.sol";
+import {Hash} from "../src/core/libraries/Hash.sol";
+import {NaiveMerkle} from "./merkle/Naive.sol";
+import {MerkleLibTest} from "./merkle/MerkleLib.t.sol";
+
+contract NewOutboxTest is Test {
+  using Hash for DataStructures.L2ToL1Msg;
+
+  address internal constant ROLLUP_CONTRACT = address(0x42069123);
+  address internal constant NOT_ROLLUP_CONTRACT = address(0x69);
+  address internal constant NOT_RECIPIENT = address(0x420);
+  uint256 internal constant DEFAULT_TREE_HEIGHT = 2;
+  uint256 internal constant AZTEC_VERSION = 1;
+
+  NewOutbox internal outbox;
+  NaiveMerkle internal zeroedTree;
+  MerkleLibTest internal merkleLibTest;
+
+  function setUp() public {
+    outbox = new NewOutbox(ROLLUP_CONTRACT);
+    zeroedTree = new NaiveMerkle(DEFAULT_TREE_HEIGHT);
+    merkleLibTest = new MerkleLibTest();
```

Comment:
I think it's a bit weird to instantiate the whole foundry test file like this just to get access to `calculateTreeHeightFromSize` function so I would prefer to just create a separate file with the util function and import just that but it's not a big deal.